create/application.go

```diff
@@ -0,0 +1,277 @@
+	Env         map[string]string `help:"Environment variables which are passed to the app at runtime."`
+}
+
+const (
+	applicationNameLabel = "application.apps.nine.ch/name"
+
+	buildStatusRunning = "running"
+	buildStatusSuccess = "success"
+	buildStatusError   = "error"
+	buildStatusUnknown = "unknown"
+
+	releaseStatusAvailable      = "available"
+	releaseStatusReplicaFailure = "replicaFailure"
+)
+
+func (app *applicationCmd) Run(ctx context.Context, client *api.Client) error {
+	fmt.Println("Creating a new application")
+	newApp := app.newApplication(client.Namespace)
+	c := newCreator(client, newApp, strings.ToLower(apps.ApplicationKind))
+	ctx, cancel := context.WithTimeout(ctx, app.WaitTimeout)
+	defer cancel()
+
+	if err := c.createResource(ctx); err != nil {
+		return err
+	}
+
+	if !app.Wait {
+		return nil
+	}
+
+	if err := c.wait(
+		ctx,
+		waitForBuildStart(newApp),
+		waitForBuildFinish(newApp),
+		waitForRelease(newApp),
+	); err != nil {
+		if buildErr, ok := err.(buildError); ok {
+			buildErr.printMessage()
+		}
+		if releaseErr, ok := err.(releaseError); ok {
+			releaseErr.printMessage()
+		}
+		return err
+	}
+
+	if err := client.Get(ctx, api.ObjectName(newApp), newApp); err != nil {
+		return err
+	}
+
+	fmt.Printf("\nYour application %q is now available at:\n  https://%s\n\n", newApp.Name, newApp.Status.AtProvider.CNAMETarget)
+
+	printUnverifiedHostsMessage(newApp)
+	return nil
+}
+
+func (app *applicationCmd) newApplication(namespace string) *apps.Application {
+	name := getName(app.Name)
+	size := apps.ApplicationSize(app.Size)
+	return &apps.Application{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: apps.ApplicationSpec{
+			ResourceSpec: runtimev1.ResourceSpec{
+				WriteConnectionSecretToReference: &runtimev1.SecretReference{
+					Name:      name,
+					Namespace: namespace,
+				},
+			},
+			ForProvider: apps.ApplicationParameters{
+				Git: apps.ApplicationGitConfig{
+					GitTarget: apps.GitTarget{
+						URL:      app.GitURL,
+						SubPath:  app.GitSubPath,
+						Revision: app.GitRevision,
+					},
+				},
+				Hosts: app.Hosts,
+				Config: apps.Config{
+					Size:     &size,
+					Replicas: pointer.Int32Ptr(app.Replicas),
+					Port:     pointer.Int32Ptr(app.Port),
+					Env:      toEnvVars(app.Env),
+				},
+			},
+		},
+	}
+}
+
+func toEnvVars(env map[string]string) apps.EnvVars {
+	vars := apps.EnvVars{}
+	for k, v := range env {
+		vars = append(vars, apps.EnvVar{Name: k, Value: v})
+	}
+	return vars
+}
+
+type buildError struct {
+	build *apps.Build
+}
+
+func (b buildError) Error() string {
+	return fmt.Sprintf("build failed with status %s", b.build.Status.AtProvider.BuildStatus)
+}
+
+func (b buildError) printMessage() {
+	fmt.Printf("\nYour build has failed with status %q. To view the build logs run the command:\n",
+		b.build.Status.AtProvider.BuildStatus)
+	fmt.Printf("  nctl logs build %s\n\n", b.build.Name)
```

Comment:
This is just a placeholder, the command might end up being different. It would also be kind of cool if it could show the last x (10?) log lines of the build as chances are good they might already contain the problem.

Reply:
That would be really cool, if it had last 10 lines plus that call to action to view all the logs it would be a really nice developer experience

Reply:
I added this to the builds CLI ticket, it should be straightforward to add once we have logging built in.